### ISSUE TITLE
SGE automatically disable compute node queue when maintain_initial_size is on

### ISSUE THREAD

**Issue by elipapa** [awslabs/cfncluster#81]
I have set up a cfncluster which I would like to leave on for a prolonged period of time. I set it up with spot instances, but I'd like to keep one compute node up in order to reduce startup times for new jobs.
Using the following settings works:
```
cluster_type = spot
initial_queue_size = 1
maintain_initial_size = true
```
However after a certain period of empty queue, the SGE queue associated with the "persistent" compute node is automatically disabled. A `qstat -f` shows for instance:

```
queuename                      qtype resv/used/tot. load_avg arch          states
---------------------------------------------------------------------------------
all.q@ip-172-31-44-86.ec2.inte BIP   0/0/4          0.01     lx-amd64      d
```

Any new jobs I submit at this stage using `qsub` remain queued for a certain period of time and then trigger the initialisation of a _new_ compute node. The existing node remains unused.

The only way to activate it again is to run 

`sudo /opt/sge/bin/lx-amd64/qmod -e all.q@ip-172-31-44-86.ec2.internal`

which is not highly desirable since it requires sudo. I was hoping I could avoid giving sudo privileges to the users of this cluster.

I have looked in the sge documentation, online and in the config files peppered throughout `/opt/sge` for a potential flag or option causing this behaviour, but I could not find any hint. 

How can I prevent this behavior?

**Comment by dougalb** [comment]
Hi, it looks like it is an unexpected behaviour of #68 and hosted are being locked but not unlocked because they should not be terminated due to `maintain_initial_size` being set. Taking a look.

**Comment by elipapa** [comment]
Thanks! Also worth looking at giving the `ubuntu/ec2-user`  rights over the administration of the queues (ie sge-admin rights). Have to recur to the slightly awkward:

     sudo SGE_ROOT=/opt/sge /opt/sge/bin/lx-amd64/qmod -e all.q@ip-172.31.44.86.ec2.internal

to re-enable the queue, since `ec2-user` does not have admin rights over the queue and it seems `root` does not have SGE_ROOT exported or a working gridengine setup.

I have also tried to `sudo sgeadmin` but it requires a password i think.

**Comment by dougalb** [comment]
The `sgeadmin` and `root` user both have admin rights for SGE. You need to call sudo with the `-i` flag to get the profile loaded. Both the following will work:

`sudo -i qmod -e all.q@ip-172.31.44.86.ec2.internal`
or
`sudo -u sgeadmin -i qmod -e all.q@ip-172.31.44.86.ec2.internal`

Just completing some internal testing of a patch for `nodewatcher` that addresses the disabled nodes.

**Comment by dougalb** [comment]
Released as part of 1.1
